### DISCUSSION TITLE
Task Group Execution Timeout

### DISCUSSION THREAD

**Discussion by seanmuth** [apache/airflow#36352] [Ideas]
Would be useful to implement an execution timeout that applies to all tasks in a task group cumulatively.

For example:

A DAG contains a Task Group with potentially 100s of mapped tasks. The user wants a timeout of 8hrs for the entire task group to complete. Due to concurrency and resource restraints, not all tasks in the group are executed concurrently, starting at different times. Setting `execution_timeout` of 8h on the mapped tasks themselves will not achieve the desired result, and it's easily possible for the task group to run for days, considering low concurrency, high task duration, and retries upon failures.

If the "Overall Duration" concept that exists in the Web UI views could be extended into the scheduler in some way, then task group execution timeouts could be implemented.

**Comment by potiuk** [comment]
I think it could be the next step after implementing task-based SLAs. https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-57+Refactor+SLA+Feature

The effort to implement it is stalled a bit, but hopefully we can re-start it soon and this could be one of next steps.
